lib/graph3d/Settings.js

```diff
@@ -488,11 +488,27 @@ function setSurfaceColor(surfaceColors, dst) {
   let rgbColors = [];
 
   if(Array.isArray(surfaceColors)) {
-    rgbColors = surfaceColors.map(util.hexToRGB);
+    // https://stackoverflow.com/questions/8027423/how-to-check-if-a-string-is-a-valid-hex-color-representation/8027444
+    const hexTestRegex = /^#([0-9A-F]{3}){1,2}$/i;
+    rgbColors = surfaceColors.map(function(colorCode){
+      if(!hexTestRegex.test(colorCode)) {
```

Comment:
Could you use `util.isValidHex()`, please? We don't want to have color matching regexes hardcoded all over the place.

Reply:
Oh didn't know that was a function thank you!